app/server/controllers/taskList.ts

```diff
@@ -0,0 +1,56 @@
+const appInsights = require('app/server/app-insights');
+const moment = require('moment');
+import { Router } from 'express';
+const paths = require('app/server/paths');
+
+const DEADLINE_EXPIRY_DATE_FORMAT = 'D MMMM YYYY';
+
+function processDeadline(expiryDateRaw: any, allQuestionsSubmitted: any) {
+  if (allQuestionsSubmitted) {
+    return { status: 'completed', formatted: null, extendable: false };
+  }
+  const expiryDate = moment.utc(expiryDateRaw);
+  const formatted = expiryDate.format(DEADLINE_EXPIRY_DATE_FORMAT);
+  const endOfToday = moment().utc().endOf('day');
+  let status = 'pending';
+  if (expiryDate.isBefore(endOfToday)) {
+    status = 'expired';
+  }
+  return { status, formatted, extendable: true };
+}
+
+const getSubmittedQuestionCount = (questions: any) => questions.filter((q: any) => q.answer_state === 'submitted').length;
+
+function getTaskList(getAllQuestionsService: any) {
+  return async(req: any, res: any, next: any) => {
```

Comment:
I'm guessing you're going to set types later, but the Express definition has these, `Request`, `Response` and `NextFunction`